client/src/templates/Challenges/redux/create-question-epic.js

```diff
@@ -50,6 +50,27 @@ function createQuestionEpic(action$, state$, { window }) {
         block,
         helpCategory
       } = challengeMetaSelector(state);
+      if (
+        challengeFiles?.some(file => file.editableRegionBoundaries.length > 0)
+      ) {
+        const editableRegionStrings = fileExtension => {
+          const startComment = fileExtension === 'html' ? '<!--' : '/*';
+          const endComment = fileExtension === 'html' ? '-->' : '*/';
+          return `\n${startComment} 'User Editable Region' ${endComment}\n`;
+        };
+
+        challengeFiles.forEach(file => {
+          const { contents, editableRegionBoundaries, ext } = file;
+          if (editableRegionBoundaries.length > 0) {
+            const comment = editableRegionStrings(ext);
+            const [start, end] = editableRegionBoundaries;
+            const editableContents = contents.split('\n');
+            editableContents.splice(start, 0, comment);
+            editableContents.splice(end, 0, comment);
+            file.contents = editableContents.join('\n');
```

Comment:
Sorry, I didn't spot this before, but we can't mutate properties of `challengeFiles`.  If we do that it mutates the redux store.  To see the impact

1. Click the ask for help button on, say, http://localhost:8000/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-2
2. Hide the editor by clicking on `index.html`
3. Re-open the editor and see

![image](https://user-images.githubusercontent.com/15801806/204343743-b8fc7b06-b728-472f-a07f-aab4a640f14a.png)

The quick fix is to re-write this to use a clone of `challengeFiles`, but the nicer fix is to avoid mutation entirely.

Reply:
Thank you for demonstrating the problem so clearly. Ive fixed it accordingly.